gapis/api/transform2/transform.go

```diff
@@ -23,6 +23,10 @@ import (
 	"github.com/google/gapid/gapis/api"
 )
 
+// StateMutator is a function that allows transformation to mutate state during transformation
+// TODO: Refactor the transforms that uses this to remove this behaviour
+type StateMutator func(id api.CmdID, cmds []api.Cmd) error
```

Comment:
This function should not take an "id" argument. That should be stored inside the TransformChain so that a transform cannot change the ID when it performs a state mutation. This design was created in-part to standardise command id generation to a single source of truth within the program and the argument here undoes that aspect of the design. See the implementation of "func (chain *TransformChain) stateMutator" for how a Transform currently transforming commands for ID=5 may pass its output on to a next transform in the chain under an incorrect ID of say 66.